### PR TITLE
Serialise SSH Public Key

### PR DESCRIPTION
**Pull request by carolinacass**
Motivation:
Needed a function function that takes a NIOSSHPublicKey and turns it into OpenSSH public key string

### CODE REVIEW

File: Sources/NIOSSH/Keys And Signatures/NIOSSHPublicKey.swift [hunk 1]
```diff
@@ -56,6 +56,17 @@ public struct NIOSSHPublicKey: Hashable {
         self = key
     }
 
+
+    /// Takes a NIOSSHPublicKey and turns it into OpenSSH public key string
+    /// - Returns: OpenSSH public key string in the format of "algorithm-id base64-encoded-key"
+    public func keyIntoComponents() -> String {
```
Comment (Lukasa) [inline Sources/NIOSSH/Keys And Signatures/NIOSSHPublicKey.swift:62]: So we should probably spell this as:

```swift
extension String {
    public init(openSSHPublicKey: NIOSSHPublicKey)
}
```

**Review by Lukasa — COMMENTED** [review]
Generally looks really good! One minor nit on the API spelling, but otherwise this is fantastic.

File: Sources/NIOSSH/Keys And Signatures/NIOSSHPublicKey.swift [hunk 2]
```diff
@@ -443,3 +443,13 @@ extension ByteBuffer {
         return returnValue
     }
 }
+extension String {
+    /// Takes a NIOSSHPublicKey and turns it into OpenSSH public key string in the format of "algorithm-id base64-encoded-key"
+    public init(openSSHPublicKey: NIOSSHPublicKey){
```
Comment (Lukasa) [inline Sources/NIOSSH/Keys And Signatures/NIOSSHPublicKey.swift:448]:
```suggestion
    public init(openSSHPublicKey: NIOSSHPublicKey) {
```

**Review by Lukasa — APPROVED** [review]

**Comment by argon** [comment]
🎉

**Pull request merged**